src/stream/stream.client.ts

```diff
@@ -274,16 +285,17 @@ export class StreamClient extends EventEmitter<StreamClientEvents> {
    *
    * @param {Error} error
    */
-  private onError(error: Error): void {
-    this.emit('error', error);
+  private onError(event: WebSocket.ErrorEvent): void {
+    this.emit('error', event.error);
   }
 
   /**
    * If a connection exists cleanup listeners
    */
   private cleanupConnection(): void {
-    this.connection?.removeAllListeners();
-    this.connection?.on('error', () => null);
+    if (!this.connection) return;
+    this.connection.removeAllListeners();
```

Comment:
I don't think this is supported. I think the event listeners we set have to be replaced with `null` or `delete` will have to be used. I would say the first as you can chain them.

Reply:
```suggestion
    this.connection.onopen = this.connection.onclose = this.connection.onmessage = null;
```

Reply:
Hoping to clarify before pushing this change:

Here's the code in `master` currently:
```ts
  private cleanupConnection(): void {
    this.connection?.removeAllListeners();
    this.connection?.on('error', () => null);
  }
```

My understanding of `isomorphic-ws`'s API is that it does not support `.on('error', func)` and uses `.onerror = func` instead. Trying that directly properly warns that `this.connection` is possibly `undefined`. If I try to follow the existing pattern with the optional chaining operator `?.` I correctly get warned about it not being value for the left hand of an assignment (`The left-hand side of an assignment expression may not be an optional property access`). I figured adding a `!this.connection` guard for the undefined case is equivalent to the existing behavior.

Are you suggesting the change of:
  1) Removing the already-existing call to `removeAllListeners` entirely, and
  2) Setting `onopen`, `onclose`, and `onmessage` to `null` in its place (and not `() => null` like we do `onerror`)?